spec.bs

```diff
@@ -1192,41 +1194,137 @@ partial interface Navigator {
+            <thead><tr><th>Group member</th><th>Interest group field</th></tr></thead>
+            <tr>
+              <td>"biddingLogicUrl"</td>
+              <td>[=interest group/bidding url=]</td>
+            </tr>
+            <tr>
+              <td>"biddingWasmHelperUrl"</td>
+              <td>[=interest group/bidding wasm helper url=]</td>
+            </tr>
+            <tr>
+              <td>"dailyUpdateUrl"</td>
+              <td>[=interest group/daily update url=]</td>
+            </tr>
+            <tr>
+              <td>"trustedBiddingSignalsUrl"</td>
+              <td>[=interest group/trusted bidding signals url=]</td>
+            </tr>
+          </table>
+          1. Let |parsedUrl| be the result of running the [=URL parser=] on |value|.
+          1. [=iteration/continue=] if any of the following conditions hold:
+            * |key| is not |groupMember|;
+            * |parsedUrl| is an error;
+            * |parsedUrl| is not [=same origin=] with |ig|'s [=interest group/owner=];
+            * |parsedUrl| [=includes credentials=];
+            * |parsedUrl| [=url/fragment=] is not null.
+          1. Set |ig|'s |interestGroupField| to |parsedUrl|.
+          
+        <dt>"<code>trustedBiddingSignalsKeys</code>"
+        <dd>
+        1. If |value| is a [=list=] of [=strings=],
+          set |ig|'s [=interest group/trusted bidding signals keys=] to |value|.
+
+        <dt>"<code>ads</code>"
+        <dt>"<code>adComponents</code>"
+        <dd>
+        1. For each |groupMember| and |interestGroupField| in the following table <table class="data">
+            <thead><tr><th>Group member</th><th>Interest group field</th></tr></thead>
+            <tr>
+              <td>"ads"</td>
+              <td>[=interest group/ads=]</td>
+            </tr>
+            <tr>
+              <td>"adComponents"</td>
+              <td>[=interest group/ad components=]</td>
+            </tr>
+          </table>
+          1. If |key| is |groupMember| and |value| is a [=list=] of {{AuctionAd}},
+            [=list/for each=] |ad| of |value|:
+            1. Let |igAd| be a new [=interest group ad=].
+            1. Let |renderUrl| be the result of running the [=URL parser=] on
+              |ad|["{{AuctionAd/renderUrl}}"].
+            1. [=iteration/continue=] if any of the following conditions hold:
+              * |renderUrl| is an error;
+              * |renderUrl| [=url/scheme=] is not "`https`";
+              * |renderUrl| [=includes credentials=].
+            1. Set |igAd|'s [=interest group ad/render url=] to |renderUrl|.
+            1. If |ad|["{{AuctionAd/metadata}}"] [=map/exists=], then let
+              |igAd|'s [=interest group ad/metadata=] be the result of
+              [=serializing a JavaScript value to a JSON string=], given |ad|["{{AuctionAd/metadata}}"].
+              This can [=exception/throw=] a {{TypeError}}.

```

Comment:
What should we do with exceptions in this case? I guess this points to a larger problem where invoking "serializing a JavaScript value to a JSON string=] technically feels invalid here, because we're "in parallel" (IIUC) without any JS execution context on the stack. Off the top of my head I'm not sure how we fix that, so maybe for now we can just specify exactly how we handle errors here, because we certainly can't re-throw them since we're not re-throwing them to any script. Specifying how this impacts the iteration would be good.

Reply:
I reworked what happens when any part of the update fails.